src/core/thread/dua_manager.hpp

```diff
@@ -129,17 +149,113 @@ class DuaManager : public InstanceLocator
     void Restore(void);
+#endif
+
+#if OPENTHREAD_CONFIG_TMF_PROXY_DUA_ENABLE
+    void UpdateChildDomainUnicastAddress(const Child &aChild, Mle::ChildDuaState aState);
+#endif
 
 private:
+    enum
+    {
+        kNewRouterRegistrationDelay = 5,    ///< Delay (in seconds) for waiting link establishment for a new Router.
+        kStateUpdatePeriod          = 1000, ///< 1000ms period  (i.e. 1s)
+    };
+
+#if OPENTHREAD_CONFIG_DUA_ENABLE
     otError GenerateDomainUnicastAddressIid(void);
     otError Store(void);
 
+    void AddDomainUnicastAddress(void);
+    void RemoveDomainUnicastAddress(void);
+    void UpdateRegistrationDelay(uint8_t aDelay);
+#endif
+
+#if OPENTHREAD_CONFIG_TMF_PROXY_DUA_ENABLE
+    void SendAddressNotification(Ip6::Address &aAddress, ThreadStatusTlv::DuaStatus aStatus, const Child &aChild);
+#endif
+
+    static void HandleNotifierEvents(Notifier::Receiver &aReceiver, Events aEvents)
+    {
+        static_cast<DuaManager &>(aReceiver).HandleNotifierEvents(aEvents);
+    }
+
+    void HandleNotifierEvents(Events aEvents);
+
+    static void HandleTimer(Timer &aTimer);
+    void        HandleTimer(void);
+
+    static void HandleRegistrationTask(Tasklet &aTasklet);
+    void        ScheduleTimer(void);
+
+    static void HandleDuaResponse(void *               aContext,
+                                  otMessage *          aMessage,
+                                  const otMessageInfo *aMessageInfo,
+                                  otError              aResult)
+    {
+        static_cast<DuaManager *>(aContext)->HandleDuaResponse(
+            *static_cast<Coap::Message *>(aMessage), *static_cast<const Ip6::MessageInfo *>(aMessageInfo), aResult);
+    }
+
+    void HandleDuaResponse(Coap::Message &aMessage, const Ip6::MessageInfo &aMessageInfo, otError aResult);
+
+    static void HandleDuaNotification(void *aContext, otMessage *aMessage, const otMessageInfo *aMessageInfo)
+    {
+        static_cast<DuaManager *>(aContext)->HandleDuaNotification(
+            *static_cast<Coap::Message *>(aMessage), *static_cast<const Ip6::MessageInfo *>(aMessageInfo));
+    }
+
+    void HandleDuaNotification(Coap::Message &aMessage, const Ip6::MessageInfo &aMessageInfo);
+    void ProcessDuaResponse(Coap::Message &aMessage);
+
+    void PerformNextRegistration(void);
+    void UpdateReregistrationDelay(void);
+    void UpdateCheckDelay(uint8_t aDelay);
+
+    TimerMilli     mTimer;
+    Tasklet        mRegistrationTask;
+    Coap::Resource mDuaNotification;
+
+    bool mIsDuaPending : 1;
+
+#if OPENTHREAD_CONFIG_DUA_ENABLE
+    enum DuaState
+    {
+        kNotExist,    ///< DUA is not avaiable.
+        kToRegister,  ///< DUA is to be registered.
+        kRegistering, ///< DUA is being registered.
+        kRegistered,  ///< DUA is registered.
+    };
+
+    DuaState  mDuaState : 2;
+    uint8_t   mDadCounter;
+    TimeMilli mLastRegistrationTime; ///< The time (in milliseconds) when sent last DUA.req or received DUA.rsp.
```

Comment:
move `TimeMilli mLastRegistrationTime; ` up for better byte alignment?

Reply:
up to where?
Intented to let `mDuaState:2` closely after `mIsDuaPending : 1;`, maybe here could move `TimeMilli mLastRegistrationTime; ` after below `mFixedDuaInterfaceIdentifier` which is in fact a byte array.

Reply:
Putting before `DuaState`  can have better alignment ? Anyway it's a nit.